src/ui/ResponsiveComponents/ResponsiveTabs.ts

```diff
@@ -249,21 +246,37 @@ export class ResponsiveTabs implements IResponsiveComponent {
     this.dropdownHeader.removeClass('coveo-dropdown-header-active');
   }
 
-  private addToDropdown(el: HTMLElement) {
-    if (this.dropdownContent) {
-      $$(el).addClass('coveo-tab-dropdown');
-      const list = this.dropdownContent.find('ol');
-      const listElement = $$('li');
-      listElement.el.appendChild(el);
-      $$(<HTMLElement>list).prepend(listElement.el);
+  private addToDropdown(tab: HTMLElement) {
+    if (!tab) {
+      return;
+    }
+    if (this.tabIsInDropdown(tab)) {
+      return;
+    }
+    if (!this.dropdownHeader) {
+      return;
     }
```

Comment:
I think it would be cleaner to extract this to a separate method
```
private canAddDropdown(tab: HTMLElement) {
  return tab && !this.tabIsInDropdown(tab) && this.dropdownHeader;
}
```
and then
```
if (!canAddDropdown) {
  return;
}
```
well something like that 😄

Reply:
Dunno, there's 2 philosophy : 

* Make sure everything is perfect upstream before calling a module/function, and then have it crash when it's not done properly. The function itself will stay simpler.
* Make the function as reliable as possible and able to recover correctly/automatically when possible, but the function/method itself gets more complicated.

In that case, this means that upstream I don't have to check if I'm adding a tab that's already present in the dropdown, or was removed in any shape or form, I can just throw anything at it and it will do the "correct" thing.

In this precise case, think if you are resizing the window up and down very fast, it's going to add/remove tabs in the dropdown, or remove a tab from a dropdown but the tab is too big and the tab section starts overflowing again, so we try to add it back etc. So we'd have to ensure this every time we call this function.

In this case, it seems to me that there's way less possibility of a bug if the "addDropdown" method is reliable.

I can certainly modify the function name though.